src/text_file_functions.cpp

```diff
@@ -316,7 +317,80 @@ QString AOApplication::get_stylesheet(QString target_tag, QString p_file)
   return f_text; // This is the empty string if no appends took place
 }
 
-QVector<QStringList> AOApplication::get_highlight_color()
+QMap<Color, QStringList> AOApplication::get_chatmessage_colors()
+{
+  QMap<Color, QStringList> default_colors;
+  default_colors[CWhite] = QStringList{"White", "#D5D5D5"};
+  default_colors[CGreen] = QStringList{"Green", "#65C856"};
+  default_colors[CRed] = QStringList{"Red", "#BA1518"};
+  default_colors[COrange] = QStringList{"Orange", "#D55900"};
+  default_colors[CBlue] = QStringList{"Blue", "#1588C8"};
+  default_colors[CYellow] = QStringList{"Yellow", "#E7CE4E"};
+  default_colors[CPurple] = QStringList{"Purple", "#F776FD"};
+  default_colors[CPink] = QStringList{"Pink", "#DA7C80"};
+
+  // File lookup order
+  // 1. In the theme folder (gamemode-timeofday/main/default), look for
+  // "courtroom_text_colors.ini".
+
+  QString path = find_theme_asset_path("courtroom_text_colors.ini");
+  if (path.isEmpty())
+    return default_colors;
+
+  QSettings text_colors(path, QSettings::IniFormat);
+
+  // Make copy for new colors
+  QMap<Color, QStringList> read_colors;
+  foreach (Color color, default_colors.keys())
+  {
+    QStringList color_parameters = default_colors[color];
+    read_colors[color] = color_parameters;
+  }
+
+  foreach (QString color_name, text_colors.childGroups())
+  {
+    // Capitalize because inis may not necessarily capitalize, but DRO does
+    color_name = color_name[0].toUpper() + color_name.mid(1).toLower();
```

Comment:
The opposite would be faster (and easier): using `toLower()` for matching against a lowered name makes it easier to remember: you never have to worry about capitalizing, only lowering.